libsolidity/analysis/TypeChecker.cpp

```diff
@@ -218,6 +219,49 @@ void TypeChecker::checkContractDuplicateFunctions(ContractDefinition const& _con
 	}
 }
 
+void TypeChecker::checkContractDuplicateEvents(ContractDefinition const& _contract)
+{
+	/// Checks that two events with the same name defined in this contract have different
+	/// argument types
+	map<string, vector<EventDefinition const*>> events;
+	for (EventDefinition const* event: _contract.events())
+		events[event->name()].push_back(event);
+
+	for (auto const& it: events)
```

Comment:
Wondering if we could pull out this loop into a helper because it is almost identical with `checkContractDuplicateFunctions`?

Reply:
I didn't want to make unnecessary abstractions, since it's only repeated once and someone might want to add more code inside the loop in the future.